searx/engines/mullvad_leta.py

```diff
@@ -0,0 +1,200 @@
+    headers['Accept-Encoding'] = "gzip, deflate, br"
+    # headers['Accept-Language'] =  "en-US,en;q=0.5"
+    headers['Content-Type'] = "application/x-www-form-urlencoded"
+    headers['DNT'] = "1"
+    headers['Host'] = "leta.mullvad.net"
+    headers['Origin'] = "https://leta.mullvad.net"
+    headers['Referer'] = "https://leta.mullvad.net/"
+    headers['Sec-Fetch-Dest'] = "document"
+    headers['Sec-Fetch-Mode'] = "navigate"
+    headers['Sec-Fetch-Site'] = "same-origin"
+    headers['Sec-Fetch-User'] = "?1"
+    headers['Sec-GPC'] = "1"
+    headers['TE'] = "trailers"
+    headers['Upgrade-Insecure-Requests'] = "1"
+    return headers
+
+
+def request(query: str, params: dict):
+    """pre-request callback
+    params<dict>:
+      method  : POST/GET
+      headers : {}
+      data    : {} # if method == POST
+      url     : ''
+      category: 'search category'
+      pageno  : 1 # number of the requested page
+    """
+
+    def get_country(eng_traits: EngineTraits) -> str:
+        sxng_locale = params.get('searxng_locale', 'all')
+        country = eng_traits.get_region(sxng_locale, eng_traits.all_locale)
+        return country if country is str else ''
+
+    country = get_country(traits)
+    params['url'] = search_url
+    params['method'] = 'POST'
+    params['data'] = {
+        "q": query,
+        "gl": country,
+    }
+    # pylint: disable=undefined-variable
+    if use_cache:
+        params['data']['oc'] = "on"
+    # pylint: enable=undefined-variable
+
+    if params['time_range'] in time_range_dict:
+        params['dateRestrict'] = time_range_dict[params['time_range']]
+    else:
+        params['dateRestrict'] = ''
+
+    if params['pageno'] > 1:
+        #  Page 1 is n/a, Page 2 is 11, page 3 is 21, ...
+        params['data']['start'] = ''.join([str(params['pageno'] - 1), "1"])
+
+    if params['headers'] is None:
+        params['headers'] = {}
+
+    assign_headers(params['headers'])
+    return params
+
+
+def response(resp: Response):
+    """post-response callback
+    resp: requests response object
+    """
+
+    def extract_result(dom_result: html.HtmlElement):
+        [a_elem, h3_elem, p_elem] = eval_xpath_list(dom_result, 'div/div/*')
+        return {
+            'url': extract_text(a_elem.text),
+            'title': extract_text(h3_elem),
+            'content': extract_text(p_elem),
+        }
+
+    # def are_more_pages(dom: html.HtmlElement) -> bool:
+    #    previous and next buttons. If only 1, then just previous
+    #    return len(eval_xpath_list(dom, '//main/div[3]/form')) == 2
+    # def get_page_x_of_y(dom: html.HtmlElement) -> Tuple[int, int]:
+    #    page_x_of_y = extract_text(eval_xpath(dom.body, '//main/div[3]/span'))
+    #    if page_x_of_y is None or page_x_of_y == '':
+    #        return (0, 0)
+    #    [_, x, _, y] = page_x_of_y.split(' ')
+    #    return (int(x), int(y))
```

Comment:
This code seems no longer needed, remove it.

Reply:
So this is what I was trying to find out - I can get the remaining number of pages available for Leta, so SearxNG doesn't show more pages available than expected. Looking at the codebase I couldn't find if this was a feature or not, so I commented it out.

In the meantime, I'll remove it, as it doesn't seem strictly necessary.